### PR TITLE
Add initial support for foldable devices

### LINKED ISSUE #19: Foldable support
Supporting foldables properly would mean showing comments on the right pane in a manner similar to how tablets are currently handled. Adding this support requires quite a deep dive into the architecture of the app but would be nice to have.

### PR DESCRIPTION
**Pull request by tgayle**
Adds initial support for a master-detail view on foldable devices using [activity embedding](https://developer.android.com/guide/topics/large-screens/activity-embedding). Configurable (and off by default) in settings. From my testing, this only impacts actual foldable devices, and tablets/phones don't see any visual difference.

I tried to make the foldable view fit inline with tablet experiences, so swiping back is disabled in CommentsActivity while in foldable-mode and a similar placeholder view is shown.

Things I'd like to point out:
- To support the preference to enable/disable foldable support, the app needs a full-restart, so that's added in `SettingsActivity`
- I got a few cryptic compilation errors when I added Jetpack WindowManager to the app, so I increased the Kotlin lib version to 1.8 to fix it.

Open to any feedback or changes you might have in mind.

screenshot 1

screenshot 2

Closes #19

### CODE REVIEW

**Comment by SimonHalvdansson** [comment]
Just tried it out in the emulator and looks very cool! The old behavior of just showing a very wide phone view was not nice at all on foldables and it seems you have really put in some work so thanks a lot for that. I do have some thoughts which I wonder how you feel about also.

- I do not get the two pane version in the 7.6" emulator in Android Studio - of course the only thing that matters is if it is active on the Galaxy Fold and Pixel Fold but do you know anything about this?
- I think a 50/50 split is more common (just by doing some googling now) in terms of width. I see it is 'window:splitRatio="0.4" ' right now in main_split_config.xml but changing it does not seem to have an effect. Do you have an opinion on this?
- When I've been in the two-pane mode and fold my phone together, I get to the main page, not the detail page. This seems to differ from some other implementations - I feel like staying at the detail is more natural. Thoughts?

**Comment by tgayle** [comment]
> I do not get the two pane version in the 7.6" emulator in Android Studio - of course the only thing that matters is if it is active on the Galaxy Fold and Pixel Fold but do you know anything about this?

I tested this out myself, it looks like devices also need to be on API 33+ and landscape mode to get the folding behavior, at least on emulators. I see an [old bug](https://issuetracker.google.com/issues/227231462) on Google's bugtracker that some emulators don't report their fold/split support correctly, so I also ended up updating my emulator before I could confirm it, but it does ultimately still work on physical devices and some emulators. I'm curious if you think there should still be a master-detail view for portrait mode folding devices though?

> I think a 50/50 split is more common (just by doing some googling now) in terms of width. I see it is 'window:splitRatio="0.4" ' right now in main_split_config.xml but changing it does not seem to have an effect. Do you have an opinion on this?

I like the look as well!

> When I've been in the two-pane mode and fold my phone together, I get to the main page, not the detail page. This seems to differ from some other implementations - I feel like staying at the detail is more natural. Thoughts?

Very odd. On both my physical devices and emulators, folding the device always stays on the detail page, where did you see that behavior?

**Comment by SimonHalvdansson** [comment]
Even with the new commit I am not getting the 50/50 split in my emulator so I think the conclusion is that my emulator is behaving very differently from your real device. I am glad to hear your device is staying in the detail view and that those things seem to be working.

Regarding the portrait dual view; it's hard to say without the device in my hand but I am under the impression that the Galaxy Fold 4 is in portrait mode by default when it is unfolded (this may very well be wrong) or at least that it is not crazy for a foldable to be in portrait when unfolded and "correctly" oriented. Therefore is seems nice to be dual-paned for those users.

In conclusion, since all of this is currently behind a setting and things are working for you I feel comfortable merging and pushing this out in an update so that some more people can experience this and perhaps come with some more feedback. Thanks for your PR and I hope you'll make another one if you come up with more improvements :)

**Comment by SimonHalvdansson** [comment]
I had to make a change in SplitChangeHandler as it was changing the background color of activities from transparent to solid today. Perhaps a check to see if the user has activated foldable support is the way to go there?
